core/display.js

```diff
@@ -258,41 +282,43 @@ Display.prototype = {
                 'type': 'flip'
             });
         } else {
-            var x, y, vx, vy, w, h;
-
-            x = this._damageBounds.left;
-            y = this._damageBounds.top;
-            w = this._damageBounds.right - x;
-            h = this._damageBounds.bottom - y;
+            let damage = { x: this._damageBounds.left,
```

Comment:
avoid creating temporary objects like this -- it's a big performance drain on fast-path functions (because some GCs don't handle it well)

Reply:
Any GC we care about though? A decent Javascript engine is pretty much a pre-requisite for noVNC anyway.

Any suggestions? We need to return multiple values from `intersect()`, so I'm not seeing a clear way to get rid of temporary objects.